thirdparty/kyaml/runfn/runfn.go

```diff
@@ -455,15 +367,48 @@ func getUIDGID(asCurrentUser bool, currentUser currentUserFunc) (string, error)
 	return fmt.Sprintf("%s:%s", u.Uid, u.Gid), nil
 }
 
-// ffp provides function filters
-func (r *RunFns) ffp(spec runtimeutil.FunctionSpec, api *yaml.RNode, currentUser currentUserFunc) (kio.Filter, error) {
+// getFunctionConfig will return the config that used in 'functionConfig' field in
+// function input. If r.FnConfigPath is provided, it will read and return it. Otherwise
+// returns an error.
+func (r *RunFns) getFunctionConfig() (*yaml.RNode, error) {
+	if r.FnConfigPath == "" {
+		return nil, fmt.Errorf("read function config from file but no path provided")
+	}
+	f, err := os.Open(r.FnConfigPath)
+	if err != nil {
+		return nil, err
```

Comment:
This error is worth wrapping `missing function config file`. Can you check what is the experience of invoking `kpt fn eval --fn-config <non-existing-path>` and tweak the errors all the way because mis-typing the command args is a common mistake people make.

I also realized, validation for presence of the file (and it being a KRM) can be done way early in the command execution.

Reply:
Add a check in the command implementation to check the existence of the config file. But I think it's ok to leave the check for valid KRM in here because it requires to read the file and parse and duplicates the code here.